homeassistant/components/epson/__init__.py

```diff
@@ -76,3 +77,28 @@ async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry):
     if unload_ok:
         hass.data[DOMAIN].pop(entry.entry_id)
     return unload_ok
+
+
+async def async_migrate_entry(hass, config_entry):
+    """Migrate old entry."""
+    _LOGGER.debug("Migrating unique_id in %s", config_entry.version)
+
+    old_unique_id = config_entry.unique_id
+    projector = hass.data[DOMAIN][config_entry.entry_id]
+    new_unique_id = await projector.get_serial_number()
```

Comment:
Do we need to check if the projector is online first or does this call handle that?